src/transformers/models/msclap/modeling_msclap.py

```diff
@@ -0,0 +1,1675 @@
+            [`PreTrainedTokenizer.__call__`] for details.
+
+            [What are input IDs?](../glossary#input-ids)
+        attention_mask (`torch.Tensor` of shape `(batch_size, sequence_length)`, *optional*):
+            Mask to avoid performing attention on padding token indices. Mask values selected in `[0, 1]`:
+
+            - 1 for tokens that are **not masked**,
+            - 0 for tokens that are **masked**.
+
+            [What are attention masks?](../glossary#attention-mask)
+        position_ids (`torch.LongTensor` of shape `(batch_size, sequence_length)`, *optional*):
+            Indices of positions of each input sequence tokens in the position embeddings. Selected in the range `[0,
+            config.max_position_embeddings - 1]`.
+
+            [What are position IDs?](../glossary#position-ids)
+        input_features (`torch.FloatTensor` of shape `(batch_size, num_channels, height, width)`):
+            Input audio features. This should be returnes by the [`ClapFeatureExtractor`] class that you can also
+            retrieve from [`AutoFeatureExtractor`]. See [`ClapFeatureExtractor.__call__`] for details.
+        return_loss (`bool`, *optional*):
+            Whether or not to return the contrastive loss.
+        output_attentions (`bool`, *optional*):
+            Whether or not to return the attentions tensors of all attention layers. See `attentions` under returned
+            tensors for more detail.
+        output_hidden_states (`bool`, *optional*):
+            Whether or not to return the hidden states of all layers. See `hidden_states` under returned tensors for
+            more detail.
+        return_dict (`bool`, *optional*):
+            Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+"""
+
+
+# Adapted from https://github.com/microsoft/CLAP/blob/59bc8446e3e9426bf4158810e572b0798a30cf4d/msclap/models/clap.py#L8
+class MSClapProjectionLayer(nn.Module):
+    def __init__(self, config: Union[MSClapAudioConfig, MSClapTextConfig]):
+        super().__init__()
+        self.config = config
+        hidden_size = config.hidden_size
+        projection_dim = config.projection_dim
+
+        self.linear1 = nn.Linear(hidden_size, projection_dim, bias=False)
+        self.linear2 = nn.Linear(projection_dim, projection_dim, bias=False)
+        self.layer_norm = nn.LayerNorm(projection_dim)
+        self.drop = nn.Dropout(config.projection_dropout_prob)
+        self.activation = F.gelu
+
+    def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
+        hidden_states1 = self.linear1(hidden_states)
+        hidden_states2 = self.activation(hidden_states1)
+        hidden_states2 = self.linear2(hidden_states2)
+        hidden_states2 = self.drop(hidden_states2)
+
+        hidden_states = self.layer_norm(hidden_states1 + hidden_states2)
+        return hidden_states
+
+
+# Adapted from transformers.models.clap.modeling_clap.ClapPreTrainedModel with Clap->MSClap
```

Comment:
(nit)

No need to add adapted from, since `ClapPreTrainedModel` doesn't do anything not standard

```suggestion
```